python_scripts/ensemble_sol_04.py

```diff
@@ -26,44 +26,57 @@
 # ```
 
 # %% [markdown]
-# Similarly to the previous exercise, create a gradient boosting decision tree
-# and create a validation curve to assess the impact of the number of trees
-# on the generalization performance of the model. Use the mean absolute error
-# to assess the generalization performance of the model.
+# Create a gradient boosting decision tree. Set the maximum depth of the individual
+# trees to `max_depth=7` and `learning_rate=0.5`. This will make the trees more
+# likely to fit the noise in the training set.
 
 # %%
 # solution
-import numpy as np
 from sklearn.ensemble import GradientBoostingRegressor
+
+gbdt = GradientBoostingRegressor(
+    max_depth=7,
+    learning_rate=0.5,
+)
+
+# %% [markdown]
+# Create a validation curve to assess the impact of the number of trees
+# on the generalization performance of the model. Evaluate the list of parameters
+# `param_range = [1, 2, 5, 10, 20, 50, 100, 200]` and use the mean absolute error
+# to assess the generalization performance of the model.
+# Try lowering the number of cross-validation folds to 3 to speed up the evaluation.
+
+# %%
+# solution
 from sklearn.model_selection import validation_curve
 
-gbdt = GradientBoostingRegressor()
-param_range = np.unique(np.logspace(0, 1.8, num=30).astype(int))
-train_scores, test_scores = validation_curve(
+param_range = [1, 2, 5, 10, 20, 50, 100, 200]
```

Comment:
Looking at the output, we could maybe remove the last values (200). With 100 iterations, we should have already a good gap.

Reply:
Done, thanks!